opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -116,6 +119,97 @@
 logger = getLogger(__name__)
 
 
+class _AgnosticContextManager(
```

Comment:
Would you mind moving the decorator implementation to a new file in [opentelemetry-api/src/opentelemetry/util](https://github.com/open-telemetry/opentelemetry-python/tree/main/opentelemetry-api/src/opentelemetry/util). This file is already quite long.